includes/class-wc-admin-reports-orders-stats-segmenting.php

```diff
@@ -0,0 +1,416 @@
+				    WHERE
+				        1=1
+						{$intervals_query['where_time_clause']}
+						{$intervals_query['where_clause']}
+						$segmenting_where
+				    GROUP BY
+				        time_interval, $product_segmenting_table.order_id, $segmenting_groupby
+				) AS $unique_orders_table
+				GROUP BY
+				    time_interval, $unique_orders_table.$segmenting_dimension_name
+				$segmenting_limit",
+			ARRAY_A
+		); // WPCS: cache ok, DB call ok, unprepared SQL ok.
+
+		$intervals_segments = $this->merge_segment_intervals_results( $segmenting_dimension_name, $segments_products, $segments_orders );
+		return $intervals_segments;
+	}
+
+	/**
+	 * Calculate segments for totals query where the segmenting property is bound to order (e.g. coupon or customer type).
+	 *
+	 * @param string $segmenting_select SELECT part of segmenting SQL query.
+	 * @param string $segmenting_from FROM part of segmenting SQL query.
+	 * @param string $segmenting_where WHERE part of segmenting SQL query.
+	 * @param string $segmenting_groupby GROUP BY part of segmenting SQL query.
+	 * @param string $table_name Name of SQL table which is the stats table for orders.
+	 * @param array  $totals_query Array of SQL clauses for intervals query.
+	 *
+	 * @return array
+	 */
+	protected function get_order_related_totals_segments( $segmenting_select, $segmenting_from, $segmenting_where, $segmenting_groupby, $table_name, $totals_query ) {
+		global $wpdb;
+
+		$totals_segments = $wpdb->get_results(
+			"SELECT
+						$segmenting_groupby
+						$segmenting_select
+					FROM
+						$table_name
+						$segmenting_from
+						{$totals_query['from_clause']}
+					WHERE
+						1=1
+						{$totals_query['where_time_clause']}
+						{$totals_query['where_clause']}
+						$segmenting_where
+					GROUP BY
+						$segmenting_groupby",
+			ARRAY_A
+		); // WPCS: cache ok, DB call ok, unprepared SQL ok.
+
+		// Reformat result.
+		$totals_segments = $this->reformat_totals_segments( $totals_segments, $segmenting_groupby );
+		return $totals_segments;
+	}
+
+	/**
+	 * Calculate segments for intervals query where the segmenting property is bound to order (e.g. coupon or customer type).
+	 *
+	 * @param string $segmenting_select SELECT part of segmenting SQL query.
+	 * @param string $segmenting_from FROM part of segmenting SQL query.
+	 * @param string $segmenting_where WHERE part of segmenting SQL query.
+	 * @param string $segmenting_groupby GROUP BY part of segmenting SQL query.
+	 * @param string $table_name Name of SQL table which is the stats table for orders.
+	 * @param array  $intervals_query Array of SQL clauses for intervals query.
+	 *
+	 * @return array
+	 */
+	protected function get_order_related_intervals_segments( $segmenting_select, $segmenting_from, $segmenting_where, $segmenting_groupby, $table_name, $intervals_query ) {
+		global $wpdb;
+		$limit_parts      = explode( ',', $intervals_query['limit'] );
+		$orig_rowcount    = intval( $limit_parts[1] );
```

Comment:
Do we need to validate $limit_parts has two elements here?

Reply:
Yup, good idea, thanks. Updated in https://github.com/woocommerce/wc-admin/pull/1286/commits/c83f9b0015474cf503276779af9730620c64d41d.